src/components/Company/DocumentSigner/SignatureForm/SignatureForm.test.tsx

```diff
@@ -0,0 +1,92 @@
+import { beforeEach, describe, expect, it, vi } from 'vitest'
+import { render, screen, waitFor } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+import { HttpResponse } from 'msw'
+import { SignatureForm } from './SignatureForm'
+import { GustoTestApiProvider } from '@/test/GustoTestApiProvider'
+import { setupApiTestMocks } from '@/test/mocks/apiServer'
+import { companyEvents } from '@/shared/constants'
+import { server } from '@/test/mocks/server'
+import { handleSignCompanyForm } from '@/test/mocks/apis/company_forms'
+
+vi.mock('@/hooks/useContainerBreakpoints/useContainerBreakpoints', async () => {
+  const actual = await vi.importActual('@/hooks/useContainerBreakpoints/useContainerBreakpoints')
+  return {
+    ...actual,
+    default: () => ['base', 'small', 'medium'],
+    useContainerBreakpoints: () => ['base', 'small', 'medium'],
+  }
+})
+
+const testForm = {
+  uuid: 'form-123',
+  name: 'Test Form',
+  status: 'not_signed',
+  form_type: 'company',
+  created_at: '2024-05-29T12:00:00Z',
+  updated_at: '2024-05-29T12:30:00Z',
+  requires_signing: true,
+}
+
+describe('SignatureForm', () => {
+  const mockOnEvent = vi.fn()
+
+  beforeEach(() => {
+    setupApiTestMocks()
+    mockOnEvent.mockClear()
+    vi.resetModules()
+
+    server.use(handleSignCompanyForm(() => HttpResponse.json(testForm)))
+  })
+
+  it('fires the correct events when form is submitted successfully', async () => {
+    const user = userEvent.setup()
```

Comment:
Is this needed? I've legit never used this with RTL before

Reply:
Yes! see docs here https://testing-library.com/docs/user-event/setup

I believe it could technically be called in a beforeEach, but that would mean sharing state across tests which could result in some unexpected behaviors

Reply:
Interesting TIL

Reply:
Interesting i've never used this before. TIL